### PR TITLE
Add LLVM for code generation proposal

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add LLVM for code generation proposal

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: docs/eBpfLlvmCodeGenerationProposal.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Clang/LLVM as code generator for eBPF-for-Windows
+
+## Overview
+
+Clang/LLVM has a much richer tool chain for generating optimal ISA specific
+instruction sequences. Proposal is to switch to LLVM to emit optimal native
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/eBpfLlvmCodeGenerationProposal.md]: Is this clearer? The idea is to use the more feature rich LLVM code generation functionality.

File: docs/eBpfLlvmCodeGenerationProposal.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Clang/LLVM as code generator for eBPF-for-Windows
+
+## Overview
+
+Clang/LLVM has a much richer tool chain for generating optimal ISA-specific
+instruction sequences. Proposal is to use LLVM to convert eBPF byte code to
+optimal native instruction sequences (replacing the current uBPF jitter).
```
Comment (dthaler) [inline docs/eBpfLlvmCodeGenerationProposal.md:7]:
```suggestion
optimal native instruction sequences (updating the current uBPF jitter).
```

**Review by dthaler — COMMENTED** [review]
As far as I know, LLVM doesn't have other mitigations like constant blinding.  This should be taken into account.   And it seems that the desire for such optimizations should not be limited to Windows so this issue may be better filed against the ubpf project and simply consumed by ebpf-for-windows rather than doing something only for windows.

**Comment by Alan-Jowett** [comment]
Filed: https://github.com/iovisor/ubpf/issues/72 as this is more general than just Windows. Will close out this PR.

**Pull request closed without merging**